CMakeLists.txt

```diff
@@ -67,6 +67,40 @@ if (F3D_USE_EXTERNAL_NLOHMANN_JSON)
   find_package(nlohmann_json REQUIRED)
 endif ()
 
+# check if we need to link with libatomic (not needed on MSVC)
+if ((NOT Windows) AND (NOT APPLE))
+	# TODO: migrate to CheckSourceCompiles in CMake >= 3.19
```

Comment:
We have a prerequisite on CMake 3.21 already so it's fine to migrate on `CheckSourceCompiles` now.